File: backend/appBackend/Dtos/GlobalPostWall/PostDTO.cs
using appBackend.Models;

namespace appBackend.Dtos.GlobalPostWall
{
    public class PostDTO
    {
        public Guid PostId { get; set; }
        public Guid UserId { get; set; } // Author ID
        public Guid GroupId { get; set; }
        public string AuthorUsername { get; set; } = string.Empty;
        public string AuthorFullName { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public int LikeCount { get; set; }
        public int CommentCount { get; set; }
        public bool IsLikedByCurrentUser { get; set; }
        public List<AttachmentDTO> Attachments { get; set; } = new List<AttachmentDTO>();
        public Poll Poll { get; set; } = new Poll();
    }

    public class CreatePostRequestDTO
    {
        public Guid GroupId { get; set; }
        public string Content { get; set; } = string.Empty;
        public List<IFormFile> Attachments { get; set; } = new List<IFormFile>(); // For file uploads
        public string? Poll { get; set; } // make it nullable if no poll
    }

    public class UpdatePostRequestDTO
    {
        public string? Content { get; set; } // Allow updating content
    }
}

File: frontend/appFrontend/src/app/models/postDto.ts
// src/app/models/dto.ts
export interface PostDTO {
    PostId: string;
    UserId: string;
    GroupId: string;
    AuthorUsername: string;
    AuthorFullName: string;
    Content: string;
    CreatedAt: Date;
    LikeCount: number;
    CommentCount: number;
    IsLikedByCurrentUser: boolean;
    Attachments: AttachmentDTO[];
    Poll: Poll;
  }
  
  export interface CreatePostRequestDTO {
    GroupId: string;
    Content: string;
    Attachments?: File[]; // Assuming File type for Attachments
    Poll?: Poll;
  }
  
  export interface UpdatePostRequestDTO {
    Content?: string;
  }
  
  export interface CommentDTO {
    CommentId: string;
    PostId: string;
    UserId: string;
    AuthorUsername: string;
    AuthorFullName: string;
    Content: string;
    CreatedAt: Date;
  }
  
  export interface CreateCommentRequestDTO {
    Content: string;
  }
  
  export interface AttachmentDTO {
    AttachmentId: string;
    FileName: string;
    FilePath: string;
    ContentType: string;
    FileSize: number;
  }
  
  export interface LikeDTO {
    LikeId: string;
    PostId: string;
    UserId: string;
    CreatedAt: Date;
  }

  export interface Poll {
    Question: string;
    Options: string[]; // Array of options
    IsGlobal: boolean;
    AllowMultipleSelections: boolean;
  }
  